test/test_autograd.py

```diff
@@ -4669,22 +4669,39 @@ def test_grad_fn_attr_bindings(self):
         out = torch.stack([a, b], dim=0)
         self.assertEqual(out.grad_fn._saved_tensors, (a, b))              # TensorList -> Tuple[Tensor]
         self.assertIsInstance(out.grad_fn._saved_tensors[0], torch.Tensor)
+        self.assertIsInstance(out.grad_fn._raw_saved_tensors[0], torch._C._autograd.SavedTensor)
         self.assertEqual(out.grad_fn._saved_dim, 0)                       # int64_t -> int
         self.assertIsInstance(out.grad_fn._saved_dim, int)
 
+        saved = out.grad_fn._raw_saved_tensors[0]
+        saved.register_hooks(lambda x: x, lambda x: x)
```

Comment:
Isn't this undefined behavior if Python still references `saved` after its C++ freed? Maybe we should delete after we register hooks here

Reply:
Yes you're right